src/resolvelib/resolvers.py

```diff
@@ -269,6 +272,89 @@ def _attempt_to_pin_criterion(self, name: KT) -> list[Criterion[RT, CT]]:
         # end, signal for backtracking.
         return causes
 
+    def _copy_all_states(self) -> list[State]:
+        """Create a copy of the all the current states"""
+        return [
+            State(
+                s.mapping.copy(),
+                s.criteria.copy(),
+                s.backtrack_causes[:],
+            )
```

Comment:
Maybe we should just implement the copy protocol on State.

Reply:
Done